### PR TITLE
chore(federation): Rename `buildFederatedSchema` -> `buildSubgraphSchema`

### PR DESCRIPTION
**Pull request by trevor-scheer**
This deprecates but continues to support usages of `buildFederatedSchema`
by exporting an alias of the existing function.

The naming of this fn has always been slightly confusing, but since the
introduction of the word `subgraph`, this seems like a very sensible rename.

### CODE REVIEW

**Comment by trevor-scheer** [comment]
A search of 'federated' within the codebase reveals a lot of documentation that could be migrated to 'subgraph' cc @StephenBarlow.

File: docs/source/api/apollo-federation.mdx [hunk 1]
```diff
@@ -7,13 +7,14 @@ api_reference: true
 
 This API reference documents the exports from the `@apollo/federation` package.
 
-## `buildFederatedSchema`
+## `buildSubgraphSchema`
+> formerly `buildFederatedSchema`
```
Comment (StephenBarlow) [inline docs/source/api/apollo-federation.mdx:11]: Recommend replacing with 

> This method was renamed from `buildFederatedSchema` in `@apollo/federation` v0.27.2 (the previous name still works but might be removed in a future release).

**Review by StephenBarlow — APPROVED** [review]
I looked at the remaining instances of "federated", and they all appear to describe an entire "federated graph" or a _supergraph_ schema, both of which feel fine. So at least from a docs perspective, think this is good!

**Comment by glasser** [comment]
Not reviewing but I love this idea.

**Review by abernix — APPROVED** [review]

**Pull request merged**